File: wpm-opencode-plugin/wpm-lib/infra/paths.ts
import { homedir } from "node:os"
import { join } from "node:path"

// Resolve the wpm server venv Python the same way install.sh lays it out
// (DATA_DIR = $XDG_DATA_HOME/wpm-system, default ~/.local/share/wpm-system).
export function resolvePythonPath(): string {
  const dataHome = process.env.XDG_DATA_HOME ?? join(homedir(), ".local", "share")
  return join(dataHome, "wpm-system", "venv", "bin", "python")
}

File: wpm-mcp-server/src/wpm_mcp_server/cli/paths.py
"""Shared paths for wpm CLI."""

from __future__ import annotations

import os
from pathlib import Path

DATA_DIR = os.environ.get("XDG_DATA_HOME", str(Path.home() / ".local" / "share" / "wpm-system"))
BIN_DIR = os.environ.get("XDG_BIN_HOME", str(Path.home() / ".local" / "bin"))
